crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -47,11 +47,39 @@ pub struct Mesh {
 /// ```
 /// # use bevy_render::mesh::{Mesh, Indices};
 /// # use bevy_render::render_resource::PrimitiveTopology;
-/// fn create_triangle() -> Mesh {
+/// fn create_simple_parallelogram() -> Mesh {
+///     // create a new mesh, add 4 vertices, each with its own position attribute (coordinate in
+///     // 3D space). for each of the points of the parallelogram. for more examples of different
+///     // shapes, see the built in examples in bevy.
 ///     let mut mesh = Mesh::new(PrimitiveTopology::TriangleList);
-///     mesh.insert_attribute(Mesh::ATTRIBUTE_POSITION, vec![[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [1.0, 1.0, 0.0]]);
-///     mesh.set_indices(Some(Indices::U32(vec![0,1,2])));
+///     mesh.insert_attribute(
+///         Mesh::ATTRIBUTE_POSITION,
+///         vec![[0.0, 0.0, 0.0], [1.0, 2.0, 0.0], [2.0, 2.0, 0.0], [1.0, 0.0, 0.0]]
+///     );
+///     // assign a UV coordinate to each vertex.
+///     mesh.insert_attribute(
+///         Mesh::ATTRIBUTE_UV_0,
+///         vec![[0.0, 1.0], [0.5, 0.0], [1.0, 0.0], [0.5, 1.0]]
+///     );
+///     // assign normals (everything points outwards)
+///     mesh.insert_attribute(
+///        Mesh::ATTRIBUTE_NORMAL,
+///        vec![[0.0, 0.0, 1.0], [0.0, 0.0, 1.0], [0.0, 0.0, 1.0], [0.0, 0.0, 1.0]]
+///     );
+///     // NOTE: for StandardMaterial to work properly with a mesh, it should have
+///     // an ATTRIBUTE_UV_0, and ATTRIBUTE_NORMAL, for proper texture and light processing.
+///     //
+///     // after defining all the vertices and their attributes, build each triangle using the
+///     // indices of the vertices that make it up in a counter-clockwise order. If the
+///     // indices' order would be clock-wise bevy would render the back and not the front.
```

Comment:
I think the details about how to make a StandardMaterial work with a mesh can either:

1) Be put under the example actually:
- add "to be rendered with a StandardMaterial" to "Example of constructing a mesh:"
- remove the "NOTE" from here and the "If the the indices order..."
- add a section bellow the code with a list: "To render correctly with a StandardMaterial, a mesh needs to:" with three points about UVs, normals, and vertex-winding order (with the precision that it's with the default Some(Face::Front) cull_mode)

2) Not have a note:
- add "to be rendered with a StandardMaterial" to "Example of constructing a mesh:" still
- remove the Note still
- change the comments above like "assign normals" to "assign normals so the lights are processed properly with the StandardMaterial"

I would prefer solution 1, straightforward example and details bellow. Explaining that other kind of materials can use other attributes.